kernel/syscall.c

```diff
@@ -196,17 +196,55 @@ static bool _trace_syscall(long n)
     // Check if syscall tracing is enabled.
     if (__myst_kernel_args.strace_config.trace_syscalls)
     {
+        // -1 represents strace syscall filter is not set
+        short filter_syscall = -1;
```

Comment:
Can we use `bool` here? Also personally I'd prefer naming it `syscall_allowed` or `allow_syscall`

Reply:
Cannot use bool here. There are three options - unset (-1), and set to 0 or 1. I tried to use the minimum required, but we need all three options.

Reply:
I changed the logic, we might not need the flag anymore. Thanks for making me look at it again.